src/DiffSharp.Core/Tensor.fs

```diff
@@ -2796,6 +2796,10 @@ type Tensor =
     /// <summary>Push the given value as part of the reverse-mode computation at the given output tensor.</summary>
     /// <param name="value">The value to apply.</param>
     member t.reversePush(value:Tensor) =
+        let check (v:Tensor,t:Tensor) = 
+            // check the shapes of the adjoints match the nodes to which they are being propagated
+            assert (Shape.canExpand v.shape t.derivative.shape || Shape.canExpand t.derivative.shape v.shape)

```

Comment:
This is making me extremely worried. Let's chat about it when we meet.

Reply:
Cool thanks for checking post hoc

The assert isn't failing for any tests we have, do you think it's too tight or too loose?

Yes let's discuss when we meet